### PR TITLE
[SPARK-36393][BUILD] Try to raise memory for GHA

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

According to the feedback from GitHub, the change causing memory issue has been rolled back. We can try to raise memory again for GA.

### Why are the changes needed?

Trying higher memory settings for GA. It could speed up the testing time.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

GA

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -174,8 +174,9 @@ jobs:
       run: |
         # Hive "other tests" test needs larger metaspace size based on experiment.
         if [[ "$MODULES_TO_TEST" == "hive" ]] && [[ "$EXCLUDED_TAGS" == "org.apache.spark.tags.SlowHiveTest" ]]; then export METASPACE_SIZE=2g; fi
```
Comment (viirya) [inline .github/workflows/build_and_test.yml:176]: Other tests don't need higher METASPACE_SIZE, so I keep smaller METASPACE_SIZE for others, only high METASPACE_SIZE for Hive test.

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -174,8 +174,9 @@ jobs:
...
-        export SERIAL_SBT_TESTS=1
-        ./dev/run-tests --parallelism 1 --modules "$MODULES_TO_TEST" --included-tags "$INCLUDED_TAGS" --excluded-tags "$EXCLUDED_TAGS"
+        # Hive and SQL tests become flaky when running in parallel as it's too intensive.
+        if [[ "$MODULES_TO_TEST" == "hive" ]] || [[ "$MODULES_TO_TEST" == "sql" ]]; then export SERIAL_SBT_TESTS=1; fi
+        ./dev/run-tests --parallelism 2 --modules "$MODULES_TO_TEST" --included-tags "$INCLUDED_TAGS" --excluded-tags "$EXCLUDED_TAGS"
```
Comment (viirya) [inline .github/workflows/build_and_test.yml:179]: Increasing parallelism.

**Comment by dongjoon-hyun** [comment]
Thank you, @viirya .

To @gengliangwang , could you check that the test time is reduced as you expected?
> I think we can try reverting this one. Higher parallelism and memory settings can reduce the test time.

**Comment by gengliangwang** [comment]
Thanks for doing this, @viirya 
@dongjoon-hyun Yes let's compare the running time with this build: https://github.com/Peng-Lei/spark/actions/runs/1093319423

**Comment by viirya** [comment]
Total duration:

https://github.com/Peng-Lei/spark/actions/runs/1093319423: 1h 33m 48s
https://github.com/viirya/spark-1/actions/runs/1094713153: 2h 37m 57s

**Comment by HyukjinKwon** [comment]
Thanks for doing this @viirya!

**Comment by HyukjinKwon** [comment]
if parallelism doesn't affect the total elapsed time much, we could just change memory back only.

**Comment by viirya** [comment]
Sounds good.

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -220,7 +220,7 @@ jobs:
       SPARK_LOCAL_IP: localhost
       SKIP_UNIDOC: true
       SKIP_MIMA: true
-      METASPACE_SIZE: 128m
+      METASPACE_SIZE: 1g
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:223]:
```suggestion
      METASPACE_SIZE: 1g
```

**Review by HyukjinKwon — COMMENTED** [review]
Looks fine to me!

**Comment by gengliangwang** [comment]
Does it seem that the run time is actually increased after raising memory? 
Before change: https://github.com/Peng-Lei/spark/actions/runs/1093319423
After change:  https://github.com/viirya/spark-1/actions/runs/1095919960a

**Comment by viirya** [comment]
I quickly compared the tests. Seems the tests run in two different GA runs are different.

**Comment by viirya** [comment]
Submitted #33642 with negligible change. It might be easier to compare.

**Comment by viirya** [comment]
That run in #33642 https://github.com/viirya/spark-1/actions/runs/1098982017: 1h 39m 40s

It is still faster than https://github.com/viirya/spark-1/actions/runs/1095919960a:  1h 54m 42s? Seems memory setting doesn't affect GA runtime a lot.

**Comment by HyukjinKwon** [comment]
I think it's fine to revert the memory change then. I would prefer to set it like 1g or 1024m

**Comment by viirya** [comment]
Ok. Seems okay to me. So this is good to go?

**Review by gengliangwang — APPROVED** [review]
LGTM

**Comment by viirya** [comment]
Thanks. Merging to master/3.2.

**Comment by viirya** [comment]
I will submit another one for 3.1. Thanks.

**Comment by HyukjinKwon** [comment]
Thanks guys!

**Comment by dongjoon-hyun** [comment]
Thank you, @viirya and all!

**Pull request closed without merging**